sources/packages/api/src/services/application/application.service.ts

```diff
@@ -675,35 +684,154 @@ export class ApplicationService extends RecordDataModelService<Application> {
+    locationId: number,
+    paginationOptions: PaginationOptions,
+    applicationStatus: ApplicationStatus,
+  ): Promise<PaginatedResults<Application>> {
     // TODO: there are two similar methods to get one and many records for the same list/details getActiveApplication and getActiveApplications. Can we use only one?
-    return this.repo
+    const activeApplicationQuery = this.repo
       .createQueryBuilder("application")
       .select([
         "application.applicationNumber",
         "application.id",
         "currentAssessment.id",
-        "application.applicationStatus",
         "offering.studyStartDate",
         "offering.studyEndDate",
         "student.id",
         "user.firstName",
         "user.lastName",
+        "application.isArchived",
+        "studentScholasticStanding.id",
       ])
       .leftJoin("application.currentAssessment", "currentAssessment")
       .leftJoin("currentAssessment.offering", "offering")
+      .leftJoin(
+        "currentAssessment.studentScholasticStanding",
+        "studentScholasticStanding",
+      )
       .innerJoin("application.student", "student")
       .innerJoin("student.user", "user")
       .where("application.location.id = :locationId", { locationId })
       .andWhere("application.applicationStatus = :applicationStatus", {
         applicationStatus: ApplicationStatus.completed,
-      })
-      .orderBy("application.applicationNumber", "DESC")
-      .getMany();
+      });
+
+    // Available applications are records that have no 'scholastic standing reported' and are not 'archived'.
+    if (applicationStatus === ApplicationStatus.available) {
+      activeApplicationQuery.andWhere(
+        new Brackets((qb) => {
+          qb.where("studentScholasticStanding.id IS NULL").andWhere(
+            "application.isArchived = :isArchived",
+            {
+              isArchived: false,
+            },
+          );
+        }),
+      );
+    }
+
+    // Unavailable applications are records that are either 'completed (scholastic standing reported but not yet archived)' or 'archived'.
+    if (applicationStatus === ApplicationStatus.unavailable) {
+      activeApplicationQuery.andWhere(
+        new Brackets((qb) => {
+          qb.where("studentScholasticStanding.id IS NOT NULL").andWhere(
+            "application.isArchived = :isNotArchived",
+            {
+              isNotArchived: false,
+            },
+          );
+        }),
+      );
+      activeApplicationQuery.orWhere(
+        new Brackets((qb) => {
+          qb.where("application.isArchived = :isArchived", {
+            isArchived: true,
+          });
+        }),
+      );
+    }
+    activeApplicationQuery.orderBy("application.applicationNumber", "DESC");
+
+    if (paginationOptions.searchCriteria) {
+      activeApplicationQuery
+        .andWhere(
+          new Brackets((qb) => {
+            qb.where(
+              "CONCAT(user.firstName,' ', user.lastName) Ilike :searchCriteria",
```

Comment:
Probably not part of this PR but can we do this name search in an indexed way?
The way that it is right now will always perform a full table scan in my understanding, correct me if I am wrong, please.
One suggestion is to create an indexed expression, as shown below (we created one already to deal with `lower(name)`);
```sql
CREATE INDEX users_full_name ON sims.users ((first_name || ' ' || last_name));
```
and change the query to use the same expression, like below:
```sql
SELECT * FROM sims.users WHERE (first_name || ' ' || last_name) = 'John Smith';
```
Source: https://www.postgresql.org/docs/current/indexes-expressional.html

Reply:
I like the suggestion. I will read up on it via the link.

Reply:
i too liked the sugesstion.

Reply:
Both the ways sounds good @andrewsignori-aot. 👍
Thanks for sharing.

Reply:
I created the tech debit to do it @henryoforeh-aot, no need to spend time on it right now: https://github.com/bcgov/sims/issues/1280